linker/shared/src/main/scala/org/scalajs/linker/backend/wasmemitter/FunctionEmitter.scala

```diff
@@ -2417,35 +2427,71 @@ private class FunctionEmitter private (
           fb += wa.RefAsNonNull
 
         case _ =>
-          genTree(expr, AnyType)
-
-          markPosition(pos)
+          def targetTpeCastableThroughJSWasmBoundary: Boolean = targetTpe match {
```

Comment:
As mentioned above: Push down into the genThroughCustomJSHelper functions? (otherwise it feels `castTo` "lies").